homeassistant/components/rainforest_eagle/sensor.py

```diff
@@ -49,14 +50,26 @@
 )
 
 
+def hwtest(cloud_id, install_code, ip_address):
+    """Try API call 'device_list' to see if target device is Legacy or Eagle-200."""
+    reader = LeagleReader(cloud_id, install_code, ip_address)
+    response = reader.post_cmd("device_list")
+    if "Error" in response and "Unknown command" in response["Error"]["Text"]:
+        return reader  # Probably a Legacy model
```

Comment:
It feels weird that we create a legacy device, send it a command that only works on newer ones and then if it errors, assume it's legacy.

You should either:
 - create a legacy device and test if a legacy command works
 - create a modern one and test if modern command works

Reply:
As far as I can tell, `eagle200_reader` doesn't have a way to elicit a response from the old device models, so we have to try legacy reader first.

There are a few features that seem to only be supported by the legacy models, but the API definitely has undocumented features.  I think a good candidate would be `get_network_info`, since it returns a model number.

@gtdiehl, could you run the this command with the Eagle-200 hardware and tell me what kind of response you give (scrubbed of secrets)?

```python
from uEagle import Eagle
 
ADDRESS = '192.168.1.123'
CLOUD_ID = '012abc'
INSTALL_CODE = '0123456789abcdef'
 
eagle = Eagle(CLOUD_ID, INSTALL_CODE, address=ADDRESS)

print(eagle.get_network_info())
```

Reply:
First, thanks for working on this. I messed around with it for a bit but gave up due to jankiness in the API and lack of time.

Since legacy seems to be what's broken and uEagle is AFAIK the legacy library here's the output from that:
```json
{
"NetworkInfo": {
"DeviceMacId": "redacted",
"InstallCode": "redacted",
"LinkKeyHigh": "redacted",
"LinkKeyLow": "redacted",
"FWVersion": "1.4.48 (6952)",
"HWVersion": "1.2.6",
"Manufacturer": "Rainforest Automation, Inc.",
"ModelId": "Z109-EAGLE",
"DateCode": "2015040124620816",
"ImageType": "0x1301",
"Protocol": "Zigbee"}
}
```

Happy to help with any other legacy testing.

Reply:
Thank you @philk, but I have a legacy model myself, so I still need a Eagle-200's response. My hope is that `get_network_info` has been removed from the new API and that will let us distinguish between models.  Same idea as the current code, but with an legacy-only command rather than a 200-only command.

Reply:
@jcalbert Here is the ouput I receive when I run `get_message()` against an Eagle-200
`{'Response': {'Text': 'Command processed', 'MainTag': 'Command', 'Command': 'get_network_info'}}`

Reply:
@gtdiehl excellent, thank you.  This should make for a pretty good hardware test.

Reply:
@gtdiehl thanks again.  Would you be willing to test this latest commit to make sure it correctly detects the EAGLE-200?